Sources/Analytics/ComScore/ComScoreTracker.swift

```diff
@@ -6,48 +6,109 @@
 
 import Combine
 import ComScore
+import CoreMedia
 import Foundation
 import Player
 
 /// Stream tracker for comScore.
 public final class ComScoreTracker: PlayerItemTracker {
-    private let streamingAnalytics = SCORStreamingAnalytics()
+    private var streamingAnalytics = SCORStreamingAnalytics()
     private var cancellables = Set<AnyCancellable>()
+    @Published private var metadata: [String: String] = [:]
 
-    public init(configuration: Void, metadataPublisher: AnyPublisher<[String: String], Never>) {}
+    public init(configuration: Void, metadataPublisher: AnyPublisher<[String: String], Never>) {
+        metadataPublisher.assign(to: &$metadata)
+    }
 
     public func enable(for player: Player) {
         streamingAnalytics.createPlaybackSession()
         streamingAnalytics.setMediaPlayerName("Pillarbox")
         streamingAnalytics.setMediaPlayerVersion(PackageInfo.version)
 
-        player.$playbackState
-            .sink { [weak self] playbackState in
-                self?.notify(playbackState: playbackState)
+        $metadata.sink { [weak self] metadata in
+            self?.updateMetadata(with: metadata)
+        }
+        .store(in: &cancellables)
+
+        Publishers.CombineLatest3(player.$playbackState, player.$isSeeking, player.$isBuffering)
+            .weakCapture(player)
+            .sink { [weak self] state, player in
+                self?.notify(playbackState: state.0, isSeeking: state.1, isBuffering: state.2, player: player)
             }
             .store(in: &cancellables)
     }
 
     public func disable() {
         cancellables = []
+        streamingAnalytics = SCORStreamingAnalytics()
     }
 
-    private func notify(playbackState: PlaybackState) {
-        let metadata = SCORStreamingContentMetadata { builder in
-            guard let builder else { return }
-            var customLabels = [String: String]()
-            AnalyticsListener.capture(&customLabels)
-            builder.setCustomLabels(customLabels)
-        }
+    private func notify(playbackState: PlaybackState, isSeeking: Bool, isBuffering: Bool, player: Player) {
+        AnalyticsListener.capture(streamingAnalytics.configuration())
+        guard !metadata.isEmpty else { return }
+        streamingAnalytics.setProperties(for: player)
+        streamingAnalytics.notifyEvent(playbackState: playbackState, isSeeking: isSeeking, isBuffering: isBuffering)
+    }
+
+    private func updateMetadata(with metadata: [String: String]) {
+        let builder = SCORStreamingContentMetadataBuilder()
+        builder.setCustomLabels(metadata)
+        let metadata = SCORStreamingContentMetadata(builder: builder)
         streamingAnalytics.setMetadata(metadata)
+    }
+}
 
+private extension SCORStreamingAnalytics {
+    private static func duration(for player: Player) -> Int {
+        player.timeRange.isValid ? Int(player.timeRange.duration.seconds * 1000) : 0
```

Comment:
I noticed that we have `seconds * 1000` several times.
What do you think if we create an extension on Numeric protocol to convert the seconds in milliseconds?

Here is my idea:

```swift
extension Numeric {
    var toMilliseconds: Self {
        self * 1000
    }
}
```

Reply:
Nice idea, thanks!